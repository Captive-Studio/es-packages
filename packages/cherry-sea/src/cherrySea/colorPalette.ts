import type { ColorPalette } from '../type.js';
import { aquaSerenity } from './colorPalette/aquaSerenity.js';
import { blueGrey } from './colorPalette/blueGrey.js';
import { pinkVivid } from './colorPalette/pinkVivid.js';
import { lightBlueVivid } from './colorPalette/lightBlueVivid.js';
import { limeGreen } from './colorPalette/limeGreen.js';
import { yellowVivid } from './colorPalette/yellowVivid.js';
import { redVivid } from './colorPalette/redVivid.js';
import { purple } from './colorPalette/purple.js';
import { goldenFizz } from './colorPalette/goldenFizz.js';

export const colorPalette = {
  aquaSerenity,
  blueGrey,
  pinkVivid,
  lightBlueVivid,
  limeGreen,
  yellowVivid,
  redVivid,
  goldenFizz,
  purple,
} satisfies Record<string, ColorPalette>;
