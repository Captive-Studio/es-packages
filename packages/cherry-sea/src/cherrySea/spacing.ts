import type { Spacing } from '../type.js';

export const spacing: Spacing = {
  0: '0px',
  25: '4px',
  50: '8px',
  75: '12px',
  100: '16px',
  150: '24px',
  200: '32px',
  300: '48px',
  400: '64px',
  600: '96px',
  800: '128px',
  1200: '192px',
  1600: '256px',
  2400: '384px',
  3200: '512px',
  4000: '640px',
  4800: '768px',
};
