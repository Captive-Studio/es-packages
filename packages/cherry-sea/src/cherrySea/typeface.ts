import { Typeface } from '../type.js';

const fallback = ['-apple-system', 'Segoe UI', 'roboto', 'noto sans', 'ubuntu', 'cantarell', 'helvetica neue'];

export const typeface = {
  brand: Typeface({
    fontFamily: ['Barlow', ...fallback],
  }),
  plain: Typeface({
    fontFamily: ['Inter', ...fallback],
  }),
};
