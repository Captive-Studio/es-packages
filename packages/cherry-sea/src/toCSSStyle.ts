import type { CSSVariables, Theme } from './type.js';

export function toCSSStyle(theme: Theme, prefix = ''): string {
  const cssVars = toCSSVariables(theme, prefix);
  const cssRules = Object.entries(cssVars)
    .map(([key, value]) => `  --${key}: ${value};`)
    .join('\n');
  return `:root {\n${cssRules}\n}\n`;
}

export function toCSSVariables(theme: Theme, prefix = ''): CSSVariables {
  return {
    ...toCSSVariablesObject(theme.spacing, `${prefix}-space`),
  };
}

function toCSSVariablesObject(obj: object, prefix = ''): CSSVariables {
  function getObjectEntries(input: object, entryPrefix = ''): Array<[string, string]> {
    return Object.entries(input).flatMap(([key, value]) => {
      if (value != null && typeof value === 'object' && !Array.isArray(value)) {
        // Appel récursif pour les objets imbriqués
        // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
        return getObjectEntries(value, `${entryPrefix}-${key}`);
      }
      // Formate le nom de la variable CSS
      const cssVarName = `${prefix}-${key}`.replaceAll('_', '-').toLowerCase();
      return [[cssVarName, Array.isArray(value) ? value.join(',') : String(value)]];
    });
  }

  return Object.fromEntries(getObjectEntries(obj, prefix));
}
