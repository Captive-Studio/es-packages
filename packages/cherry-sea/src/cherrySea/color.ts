import type { Color } from '../type.js';
import { colorPalette } from './colorPalette.js';

const white = '#fff';
const defaultError = 'red';

export const color = {
  // emphasisOnBlue: colorPalette.goldenFizz[400],

  // Primary
  primary: colorPalette.pinkVivid[500],
  onPrimary: white,
  primaryContainer: defaultError,
  onPrimaryContainer: defaultError,
  // Secondary
  secondary: defaultError,
  onSecondary: defaultError,
  secondaryContainer: defaultError,
  onSecondaryContainer: defaultError,
  // Tertiary
  tertiary: defaultError,
  onTertiary: defaultError,
  tertiaryContainer: defaultError,
  onTertiaryContainer: defaultError,
  // Error
  error: defaultError,
  onError: defaultError,
  errorContainer: defaultError,
  onErrorContainer: defaultError,

  surface: colorPalette.blueGrey[50],
  onSurface: colorPalette.blueGrey[900],

  outlineVariant: defaultError,
  onSurfaceVariant: colorPalette.blueGrey[700],

  outline: defaultError,
  white,
} satisfies Color;
