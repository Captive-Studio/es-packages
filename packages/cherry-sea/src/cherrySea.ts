import type { Theme } from './type.js';
import { spacing } from './cherrySea/spacing.js';
import { typeface } from './cherrySea/typeface.js';

/**
 * Cherry sea theme configuration
 */
export const cherrySea = {
  typeface,
  spacing,
} satisfies Theme;
