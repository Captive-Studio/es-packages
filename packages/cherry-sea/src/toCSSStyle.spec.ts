import { describe, it, expect } from 'vitest';
import { toCSSStyle, toCSSVariables } from './toCSSStyle.js';
import { cherrySea } from './cherrySea.js';

const THEME_PREFIX = 'cs';

describe(toCSSStyle, () => {
  it('matches snapshot', () => {
    expect(toCSSStyle(cherrySea, THEME_PREFIX)).toMatchSnapshot();
  });
});
describe(toCSSVariables, () => {
  it('matches snapshot', () => {
    expect(toCSSVariables(cherrySea, THEME_PREFIX)).toMatchSnapshot();
  });
});
