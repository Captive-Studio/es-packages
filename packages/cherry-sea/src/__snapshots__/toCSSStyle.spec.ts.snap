// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`cherrySea > matches snapshot 1`] = `
":root {
  --cs-space-0: 0px;
  --cs-space-25: 4px;
  --cs-space-50: 8px;
  --cs-space-75: 12px;
  --cs-space-100: 16px;
  --cs-space-150: 24px;
  --cs-space-200: 32px;
  --cs-space-300: 48px;
  --cs-space-400: 64px;
  --cs-space-600: 96px;
  --cs-space-800: 128px;
  --cs-space-1200: 192px;
  --cs-space-1600: 256px;
  --cs-space-2400: 384px;
  --cs-space-3200: 512px;
  --cs-space-4000: 640px;
  --cs-space-4800: 768px;
}
"
`;
